### PR TITLE
Fix return value of if/else statements to match other shells/POSIX.

### LINKED ISSUE #1443: isatty stdin not working
`➜ isatty --help`

>   Description
>
>   isatty tests if a file or file descriptor is a tty. The argument may be
>   in the form of a file path, device, or file descriptor number. Without
>   an argument, standard input is implied.

With *stdin* as argument:
````
➜ .oh-my-fish git:(Paulloz-plugin/sprunge) ✗ echo 'foo' | if isatty stdin
                                                     echo 'if'
                                             else
                                                     echo 'else'
                                             end
if
````

Without an argument:
````
➜ .oh-my-fish git:(Paulloz-plugin/sprunge) ✗ echo 'foo' | if isatty
                                                     echo 'if'
                                             else
                                                     echo 'else'
                                             end
else
````

Shouldn't both have the same behaviour? Installed fish from Git (today):
* 1ffa4d9 - (HEAD, origin/master, origin/HEAD, master) openssl: add instructions for using custom certs (6 days ago) <Jack Nagel>

### PR DESCRIPTION
**Pull request by geoff-codes**
Hopefully should  close #1443. Reverses some of #1061 / e632d39b. Note that the test case introduced in e632d39b does not need to be changed, as `if false ; end ; echo $status` **should** return 0, as the return value of the _condition_ should not affect return value.

### CODE REVIEW

File: parse_execution.cpp [hunk 1]
```diff
@@ -335,12 +340,7 @@ parse_execution_result_t parse_execution_context_t::run_if_statement(const parse
     /* Done */
     parser->pop_block(ib);
 
-    /* Issue 1061: If we executed, then always report success, instead of letting the exit status of the last command linger */
-    if (result == parse_execution_success)
-    {
-        proc_set_last_status(STATUS_BUILTIN_OK);
-    }
-
+    /* Otherwise, take the exist status of the job list. Reversal of #1061. */
```
Comment (jaredgrubb) [inline parse_execution.cpp]: /exist/exit/

**Comment by geoff-codes** [comment]
@jaredgrubb Thanks. Fixed and squashed.

**Comment by zanchey** [comment]
Looks ok to me, and I would like to take this for the next release, but I would like @ridiculousfish's eyes before merging.

**Comment by ridiculousfish** [comment]
Looks good to me - merged as b9bd0166b6952d715d82e045239465adfa0877f2 . Thanks!!

**Pull request closed without merging**